crates/rethnet_evm/benches/state_history.rs

```diff
@@ -69,6 +73,110 @@ fn bench_set_block_context_to_latest_snapshot(c: &mut Criterion) {
     );
 }
 
+fn prep_snapshots(
```

Comment:
Please correct me if I'm wrong, but what this does is:

1. for a varied number of `a`  accounts, `c` checkpoints, and `s` storage slots; fill the benchmark
  - The intend of this is such that we can see how `a`, `c`, and `s` influence the performance
2. In addition to 1, for a varied number of `p` snapshots, make additional modifications and checkpoints

This means that the parameters in a, c, and s don't actually influence the outcome of benchmarks related to snapshots anymore.

E.g. the "revert to earliest layer" benchmark should revert to the first stored layer in a database (created the first time `checkpoint` is called). Your code is currently reverting to the `c + 1`th layer.

Reply:
This is just a suggestion (the might be better solutions)!

One possible solution I see is to write:

Pseudocode:
```rust
fn benchmark(
  number_of_accounts: usize, // divisable by number_of_checkpoints without rounding
  number_of_checkpoints: usize, // divisable by number_of_snapshots without rounding
  number_of_snapshots: usize,
  number_of_storage_slots_per_account: usize,
) {
  let number_of_accounts_per_checkpoint = number_of_accounts / number_of_checkpoints;
  let number_of_checkpoints_per_snapshot = number_of_checkpoints / number_of_snapshots;
  
  for snapshot_idx in 0..number_of_snapshots {
    for checkpoint_idx in 0..number_of_checkpoints_per_snapshot {
      for account_idx in 0..number_of_accounts_per_checkpoint {
        // setup basic account
        
        for storage_slot_idx in 0..number_of_storage_slots_per_account {
          // setup storage slot
        }
      }
    }
  }
}
```

Reply:
At that point, you just need to add some logic to retrieve the state root of a particular checkpoint or snapshot

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/3921/commits/b4957daec0d5e96dc62f348a33ad0f36dc262164